akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -82,11 +82,28 @@ object BootstrapSetup {
 
 }
 
-abstract class ProviderSelection private (private[akka] val identifier: String)
+/**
+ * @param identifier the simple name of the selected provider
+ * @param fqcn the fully-qualified class name of the selected provider
+ */
+abstract class ProviderSelection private (private[akka] val identifier: String, private[akka] val fqcn: String) {
+
+  /** INTERNAL API */
+  @InternalApi private[akka] def hasCluster: Boolean =
```

Comment:
It's a bit unconventional to use `this match`, couldn't we just add another field `hasCluster`?

Reply:
@jrudolph Sure, where would you see the field going, Settings? I didn't feel this is fantastic but that was overridden by keeping the knowledge of is cluster within the `ProviderSelection` implementation, and keeping it `private[akka]`. I'd love a better idea. Though a match on type is the whole idea behind just matching on a string version of a FQCN to be safer.

Reply:
@jrudolph normally I'd not do a match with `this` or a match to do a boolean but in this one case I did see precedence in the existing codebase of a match on `this` in a few areas already, and testing `hasCluster` by a type vs String is the goal.

Reply:
I think he means leaving this method abstract here and implement it in the subclasses, e.g. 
```
case object Remote extends ProviderSelection("remote") {
  /** INTERNAL API */
  @InternalApi override private[akka] def hasCluster: Boolean = false
}
```

Reply:
@jrudolph I pushed your suggestion, thank you.